instrumentation/reactor/reactor-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/ContextPropagationOperatorInstrumentationTest.java

```diff
@@ -136,18 +160,25 @@ void storeContextForcesItToBecomeCurrent() {
               Span span =
                   testing.getOpenTelemetry().getTracer("test").spanBuilder("parent").startSpan();
 
-              return Mono.delay(Duration.ofMillis(1))
-                  .flatMap(
-                      unused -> {
-                        // usual trick to force this to run under new TracingSubscriber with context
-                        // written in the next call
-                        return new ExtensionAnnotationsTracedWithSpan().mono(Mono.just("Value"));
-                      })
-                  .subscriberContext(
-                      ctx ->
-                          ContextPropagationOperator.storeOpenTelemetryContext(
-                              ctx, Context.current().with(span)))
-                  .doFinally(unused -> span.end());
+              Mono<String> interim =
+                  Mono.delay(Duration.ofMillis(1))
+                      .flatMap(
+                          unused -> {
+                            // usual trick to force this to run under new TracingSubscriber with
+                            // context
+                            // written in the next call
+                            return new ExtensionAnnotationsTracedWithSpan()
+                                .mono(Mono.just("Value"));
+                          });
+              try {
+                interim =
+                    (Mono<String>)
+                        MONO_CONTEXT_WRITE_METHOD.invoke(
+                            interim, new StoreOpenTelemetryContext(Context.current().with(span)));
```

Comment:
In the previous version of this code, wasn't `Context.current().with(span)` executed as a part of the lambda? I mean it was lazy before, now it's eager